com.unity.mobile.notifications/Runtime/Android/Plugins/com/unity/androidnotifications/UnityNotificationUtilities.java

```diff
@@ -533,43 +533,47 @@ protected static Class<?> getOpenAppActivity(Context context) {
-            if (aInfo != null) {
-                String activityClassName = null;
-                String conflictingActivity = null;
-                boolean activityConflict = false;
-                for (ActivityInfo info : aInfo) {
-                    // activity alias not supported
-                    if (info.enabled && info.targetActivity == null) {
-                        if (activityClassName == null) {
-                            activityClassName = info.name;
-                        } else if (info.name.endsWith(".UnityPlayerActivity") || info.name.endsWith(".UnityPlayerGameActivity")) {
-                            if (activityClassName.endsWith(".UnityPlayerActivity") || activityClassName.endsWith(".UnityPlayerGameActivity")) {
-                                conflictingActivity = info.name;
-                                activityConflict = true;
-                                break;
-                            }
-
-                            activityClassName = info.name;
-                            activityConflict = false;
-                        } else {
-                            conflictingActivity = info.name;
-                            activityConflict = true;
-                        }
-                    }
-                }
+            if (aInfo == null) {
+                Log.e(TAG_UNITY, "Could not get package activities");
+                return null;
+            }
 
-                if (activityConflict) {
-                    Log.e(TAG_UNITY, String.format("Multiple choices for activity for notifications: %s and %s", activityClassName, conflictingActivity));
-                    return null;
-                }
+            String activityClassName = null;
+            String conflictingActivity = null;
+            boolean activityConflict = false;
+            for (ActivityInfo info : aInfo) {
+                // activity alias not supported
+                if (!info.enabled || info.targetActivity != null)
+                    continue;
 
                 if (activityClassName == null) {
-                    Log.e(TAG_UNITY, "Activity class for notifications not found");
-                    return null;
+                    activityClassName = info.name;
+                } else if (isUnityActivity(info.name)) {
+                    if (isUnityActivity(activityClassName)) {
+                        conflictingActivity = info.name;
+                        activityConflict = true;
+                        break;
+                    }
+
+                    activityClassName = info.name;
+                    activityConflict = false;
+                } else {
+                    conflictingActivity = info.name;
+                    activityConflict = true;
                 }
+            }
+
+            if (activityConflict) {
+                Log.e(TAG_UNITY, String.format("Multiple choices for activity for notifications: %s and %s", activityClassName, conflictingActivity));
```

Comment:
if there are two conflicting activities, this error will only print one, shouldn't it print all conflicting activities ?

Reply:
what settings?

Reply:
Notification settings. At the start of this method we take the activity from the manifest (put there during build) and this loop is a fallback. If you have multiple candidate activities, then you have to specify one in settings.

Reply:
That's fine and all, but I am talking about the error message being incomplete, it won't mention to you all activities?

Reply:
It's pointless to print them all out.
But maybe it's worth to change the message to:
"Multiple choices for activity, please explicitly specify fully qualified activity class in Notification Settings"

Reply:
generic message would be better, because with current implementation, it's not clear how you decide what activities to print in error message